static/app/components/searchQueryBuilder/tokens/filter/valueCombobox.tsx

```diff
@@ -480,6 +483,18 @@ function cleanFilterValue(
       }
       return value;
     }
+    case FieldValueType.PERCENTAGE: {
+      const parsed = parseFilterValuePercentage(value);
+      if (!parsed) {
+        return null;
+      }
+      // If the user passes "50%", convert to 0.5
+      if (parsed.unit) {
+        const numericValue = parseFloat(parsed.value);
+        return isNaN(numericValue) ? parsed.value : (numericValue / 100).toString();
```

Comment:
this might be a dumb question but in what scenario would we pass the `parsed.unit` condition, but still end up with `numericValue` being NaN

Reply:
Not a dumb question, I don't think this should actually be possible. Probably not necessary, I am being overly safe here